src/pemfile.rs

```diff
@@ -44,6 +103,7 @@ pub enum Item {
 ///
 /// You can use this function to build an iterator, for example:
 /// `for item in iter::from_fn(|| read_one(rd).transpose()) { ... }`
+#[cfg(feature = "std")]
 pub fn read_one(rd: &mut dyn io::BufRead) -> Result<Option<Item>, io::Error> {
```

Comment:
that sounds like a change orthogonal to this PR that should be discussed in a new PR. like, do you want to provide a conversion from `pemfile::Error` to `io::Error` to ease migration from pemfile 1.0 to 2.0?

Reply:
there's also the catch that you can't go from `io::Error` to `pemfile::Error` and functions like `read_one` produce `io::Error`.

one could add an `Io` variant to `pemfile::Error` that sits behind the "std" Cargo feature and mark the enum as non_exhaustive, I guess, so that if an end user enables the "std" feature in their top level crate that does not break `match`-es on the enum in dependencies :thinking:

Reply:
> that sounds like a change orthogonal to this PR that should be discussed in a new PR.

Fair, I'll resolve this conversation then.